src/kagglehub/config.py

```diff
@@ -144,3 +147,24 @@ def set_kaggle_credentials(username: str, api_key: str) -> None:
 def clear_kaggle_credentials() -> None:
     global _kaggle_credentials  # noqa: PLW0603
     _kaggle_credentials = None
+
+
+def get_colab_credentials() -> Optional[tuple[str, str]]:
+    # userdata access is already thread-safe after b/318732641
+    try:
+        from google.colab import userdata  # type: ignore[import]
+        from google.colab.errors import Error as ColabError  # type: ignore[import]
+    except ImportError:
+        return None
+
+    try:
+        username = userdata.get(COLAB_SECRET_USERNAME)
+        key = userdata.get(COLAB_SECRET_KEY)
+        return (username, key)
+    except userdata.NotebookAccessError:
+        logger.warning("Access to secret %s and %s are not granted.", COLAB_SECRET_USERNAME, COLAB_SECRET_KEY)
+    except userdata.SecretNotFoundError:
+        logger.warning("Access to secret %s and %s are not defined.", COLAB_SECRET_USERNAME, COLAB_SECRET_KEY)
```

Comment:
We shouldn't log a warning if the secret is not found. Users don't need credentials to access public resources for example. We don't want to show a warning because as long as users only access public resources, they will never need to set their credentials.

Reply:
@rosbo That's a good catch! I think ideally we should only read credentials for private models. However detecting a private model from client should NOT be what we can/want to do, so I disabled all the user warnings here. 

In reality, if the secrets were not GRANTED, colab will prompt users to choose to grant (http://screen/suVSMApVySvfMqN). The users don't need to do so for public models. What do you think?